lib/osutil/traversessymlink.go

```diff
@@ -32,47 +32,46 @@ func (e NotADirectoryError) Error() string {
 	return fmt.Sprintf("not a directory: %s", e.path)
 }
 
-// TraversesSymlink returns an error if base and any path component of name up to and
-// including filepath.Join(base, name) traverses a symlink.
+// TraversesSymlink returns an error and the part of path causing it, if base
+// and any path component of name up to and including name traverses a symlink,
+// is not a directory or is missing.
 // Base and name must both be clean and name must be relative to base.
-func TraversesSymlink(filesystem fs.Filesystem, name string) error {
+func TraversesSymlink(filesystem fs.Filesystem, name string) (error, string) {
```

Comment:
I haven't read the rest yet, just a small cleanliness note here. The `(error, string)` return type and `err, _ := ...` calls look *really* odd to me. I would prefer extending the error with the path information. That is, add a `Path string` to `NotADirectoryError` and `TraversesSymlinkError` that the caller can use after type assertion if interested. Or maybe reuse `os.PathError` which is essentially this...